marvel-app/coordinator/HomePageCoordinator.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  HomePageCoordinator.swift
+//  marvel-app
+//
+//  Created by Michal on 11/08/2022.
+//
+
+import Foundation
+import UIKit
+
+class HomePageCoordinator: Coordinator {
+    var navigationController: UINavigationController?
+    var homePageVC: HomePageViewController
+    
+    init(homePageVC: HomePageViewController){
+        self.homePageVC = homePageVC
+    }
+    
+    func eventOccured(with type: Event) {
+        switch type {
+        case .listElementClicked(let comic):
+            var vc: UIViewController & Coordinating = DetailsViewController(comic: comic)
+            vc.coordinator = self
+            navigationController?.pushViewController(vc, animated: true)
+            
+        }
+    }
+    
+    func start() -> UINavigationController{
+        navigationController = UINavigationController()
```

Comment:
Navigation controller should be passed from previous coordinator. Init it once in appCoordinator and always pass it to the next coordinators.